### ISSUE TITLE
`reportMissingSuperCall` should not be reported on `__init_subclass__` if there's no base class with an `__init_subclass__` method, when `reportUnsafeMultipleInheritance` is enabled

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#350]
```py
class Bar: ...

class Foo(Bar):
    def __init_subclass__(cls): # error: reportMissingSuperCall
        ...
```

`object.__init_subclass__` shouldn't count because it doesn't do anything (i think)

**Comment by DetachHead** [comment]
hmm, mypy reports this as an implicit override and requires adding the `override` decorator to the `__init_subclass__` method. should we do that too? if so, it doesn't really make sense to not report `reportMissingSuperCall` here.

also, it doesn't complain are extending a base class regardless of whether it defines an `__init_subclass__`. either way, this inconsistency is wrong.
